From 94cc0214dae6b3871b60988c76464b451a4b49dc Mon Sep 17 00:00:00 2001
From: Anton Kikin <a.kikin@tano-systems.com>
Date: Fri, 2 Apr 2021 01:58:02 +0300
Subject: [PATCH] net: bridge: Increase maximum hardware header length

Increase maximum hardware header length to maximum possible
KSZ tail tag size.

Signed-off-by: Anton Kikin <a.kikin@tano-systems.com>
---
 net/bridge/br_device.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/bridge/br_device.c b/net/bridge/br_device.c
index e682a668ce57..d1fd549972a6 100644
--- a/net/bridge/br_device.c
+++ b/net/bridge/br_device.c
@@ -422,6 +422,10 @@ void br_dev_setup(struct net_device *dev)
 			   NETIF_F_HW_VLAN_STAG_TX;
 	dev->vlan_features = COMMON_FEATURES;
 
+	/* Increase maximum hardware header length
+	 * to maximum possible KSZ tail tag size */
+	dev->hard_header_len += 6;
+
 	br->dev = dev;
 	spin_lock_init(&br->lock);
 	INIT_LIST_HEAD(&br->port_list);
-- 
2.17.1

